### ISSUE TITLE
Wait for all listeners to have stopped

### ISSUE THREAD

**Issue by danilofuchs** [EventEmitter2/EventEmitter2#312]
Is there a way to wait until there are no listeners running?

Context: We listen for SIGTERM, and shutdown services in the following order:

1. HTTP Client
2. Job Queues
3. Event Listeners (to make sure we don't loose events during shutdown)
4. DB + Cache connections

It would be nice to avoid shuting down the DB connection before all listeners are flushed

**Comment by DigitalBrainJS** [comment]
No, this class does not have that functionality as it does not store pending promises from called listeners to wait for the "drain" state. 
In simple cases, you can extend the class and decorate the emitAsync method to track returned promises. However, in a decent implementation, you will have to deal with race conditions, which is non-trivial and beyond the scope of the event emitter.
